### PR TITLE
PCHR-2594: Improve Modals for Mobile View

### PR DESCRIPTION
**Pull request by igorpavlov-zz**
## Overview

This PR adds styling to Modals on small screens, such as smartphone screens, and allows other repositories to import these styles (such as SSP Theme).

## Before

Desktop:

1d

Mobile:

1p

## After

Desktop:

2d

Mobile: 

2p

## Technical Description

A new partial file was created that contains the new styles for modals for smaller screens.

> scss/bootstrap/overrides/partials/_modals-xs-screen.scss

```css
@media (max-width: $screen-xs-min) { ... }
```

## Comments

This PR also fixes the modals' z-index positioning in Civi Admin (the modal appeared underneath the top navigation bar, now it is fixed).

### CODE REVIEW

File: scss/bootstrap/overrides/partials/_modals-xs-screen.scss [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// Modal for XS screens such as smartphones
```
Comment (AkA84) [inline scss/bootstrap/overrides/partials/_modals-xs-screen.scss]: I think this should go in the variants/ folder

File: scss/bootstrap/overrides/style/_modals.scss [hunk 2]
```diff
@@ -45,3 +45,13 @@
     top: initial !important;
   }
 }
+
+.modal-backdrop {
+  z-index: 100000 !important; // Some current elements have z-index 99999
+}
+
+.modal {
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_modals.scss:53]: Wouldn't it be better at this point to have
```css
.modal,
.modal-backdrop {
  z-index: 100001 !important;
}

**Review by AkA84 — CHANGES_REQUESTED** [review]

File: scss/bootstrap/overrides/partials/_modals-xs-screen.scss (hunk 1 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/partials/_modals-xs-screen.scss]: I actually had put it in variants first, then thought it is not really a variant... But OK.

File: scss/bootstrap/overrides/style/_modals.scss (hunk 2 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/style/_modals.scss:53]: They have to have different z-indexes. Setting same z-index can mess up the layout in some browsers that ignore DOM position.

Comment (deb1990) [reply scss/bootstrap/overrides/style/_modals.scss:53]: Could you give some example of which browsers can ignore DOM position?

File: scss/bootstrap/variants/_modals-xs-screen.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Modal for XS screens such as smartphones
+@media (max-width: $screen-xs-min) {
+
+  .modal-dialog:not(.modal-sm) {
+    margin: 0;
```
Comment (deb1990) [inline scss/bootstrap/variants/_modals-xs-screen.scss]: Keep alphabetical order

**Review by deb1990 — CHANGES_REQUESTED** [review]

File: scss/bootstrap/overrides/style/_modals.scss (hunk 2 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/style/_modals.scss:53]: As far as I remember, I've seen it in FF https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context. It is still good to ensure they have different z-indexes.

File: scss/bootstrap/variants/_modals-xs-screen.scss (hunk 3 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/variants/_modals-xs-screen.scss]: Done.

**Review by deb1990 — APPROVED** [review]

**Comment by deb1990** [comment]
@igorpavlov Approved, but needs to be approved by @AkA84 as well.

**Pull request merged**